types/immutability-helper/immutability-helper-tests.ts

```diff
@@ -1,48 +1,37 @@
 import * as update from 'immutability-helper';
 import { newContext } from 'immutability-helper';
 
-namespace TestObjectUpdate {
-  update({}, {
-    foo: {
-      bar: { $set: 'baz' }
-    }
-  });
-}
-
-namespace TestArrayUpdate {
-  update([], {
-    foo: {
-      bar: { $set: 'baz' }
-    }
-  });
-}
-
 namespace TestExtend {
   update.extend('$command', (specValue, originalValue) => originalValue);
 }
 
 namespace TestNewContext {
   update.newContext().extend('$command', (specValue, originalValue) => originalValue);
   newContext().extend('$command', (specValue, originalValue) => originalValue);
-
-  // This shouldn't compile, but we can't test negatives.
```

Comment:
Might as well leave this, it's handy documentation to whoever comes to read the tests. Would be nice if we could have proper negative tests though (https://github.com/DefinitelyTyped/DefinitelyTyped/issues/6987).